scripts/tests/twister/test_handlers.py

```diff
@@ -0,0 +1,822 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+Tests for handlers.py classes' methods
+"""
+
+import itertools
+import mock
+import os
+import pytest
+import signal
+import subprocess
+
+from serial import SerialException
+
+import twisterlib.harness
+
+from twisterlib.error import TwisterException
+from twisterlib.handlers import Handler, BinaryHandler, DeviceHandler, QEMUHandler
+
+
+@pytest.fixture
+def mocked_instance(tmp_path):
+    instance = mock.Mock()
+    testsuite = mock.Mock()
+    type(testsuite).source_dir = mock.PropertyMock(return_value='')
+    instance.testsuite = testsuite
+    build_dir = tmp_path / 'build_dir'
+    os.makedirs(build_dir)
+    type(instance).build_dir = mock.PropertyMock(return_value=str(build_dir))
+    platform = mock.Mock()
+    type(platform).binaries = mock.PropertyMock(return_value=[])
+    instance.platform = platform
+    type(instance.testsuite).timeout = mock.PropertyMock(return_value=60)
+    type(instance.platform).timeout_multiplier = mock.PropertyMock(return_value=2)
+
```

Comment:
```suggestion
type(instance).status = None
type(instance).reason = "Unknown"
```

Reply:
`type(instance).member = ...` syntax is necessary only when we want to have a `mock.PropertyMock`.
What is the reason for this change?

Reply:
> `type(instance).member = ...` syntax is necessary only when we want to have a `mock.PropertyMock`. What is the reason for this change?

as a suggestion, m.b. we'll need a mock property for some other test ?

Reply:
Sadly, once you `mock.PropertyMock` something, it is not reassignable. See example below:
```python
a = mock.Mock()
type(a).member = mock.PropertyMock(return_value='Mocked')
print(a.member)  # Prints 'Mocked'
a.member = 'Not mocked'
print(a.member)  # ! Still prints 'Mocked'
```

Because of that, `mock.PropertyMock`ing `status` and/or `reason` breaks the tests.